fab/fabfile.py

```diff
@@ -728,6 +730,18 @@ def force_update_static():
     execute(services_restart)
 
 
+def _tag_commit():
+    sh.git.fetch("origin", env.code_branch)
+    deploy_time = datetime.datetime.utcnow()
+    tag_name = "{:%Y-%m-%d_%H.%M}-{}-deploy".format(deploy_time, env.environment)
+    branch = "origin/{}".format(env.code_branch)
+    # TODO make it easier than --set message="my message"
+    msg = getattr(env, "message", "")
+    msg += "\n{} deploy at {}".format(env.environment, deploy_time.isoformat())
```

Comment:
Annotated tags require you to specify a message, so I just made a default one, and gave an optional way to be more descriptive.  You can do
`$ fab production awesome_deploy --set message="emergency bugfix"`
although I'm not sure this'll ever be used